kinds/report.go

```diff
@@ -0,0 +1,113 @@
+//go:build ignore
+// +build ignore
+
+//go:generate go run report.go
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/grafana/pkg/kindsys"
+	"github.com/grafana/grafana/pkg/registry/corekind"
+)
+
+func main() {
+	kinds := loadKinds()
+
+	sort.Slice(kinds, func(i, j int) bool {
+		return kinds[i].Name < kinds[j].Name
+	})
+
+	out := elsedie(json.MarshalIndent(kinds, "", "  "))("error generating json output")
+
+	path := filepath.Join(kindsys.DeclParentPath, "report.json")
+	file := codejen.NewFile(path, out, reportJenny{})
+	filesystem := elsedie(file.ToFS())("error building in-memory file system")
+
+	cwd := elsedie(os.Getwd())("error getting working directory")
+	groot := filepath.Dir(cwd)
+
+	if _, set := os.LookupEnv("CODEGEN_VERIFY"); set {
+		if err := filesystem.Verify(context.Background(), groot); err != nil {
+			die(fmt.Errorf("generated code is out of sync with inputs:\n%s\nrun `make gen-cue` to regenerate", err))
+		}
+	} else if err := filesystem.Write(context.Background(), groot); err != nil {
+		die(fmt.Errorf("error while writing generated code to disk:\n%s", err))
+	}
+}
+
+type kind struct {
+	Name     string `json:"name"`
+	Maturity string `json:"maturity"`
+	Type     string `json:"type"`
+	Latest   string `json:"latest,omitempty"`
+}
```

Comment:
What's the benefit of creating a new type for this vs. using the existing ones?

It seems to me that just directly printing the `CoreStructuredMeta` or `RawMeta` that can be gotten from the `corekinds.Base` elmeents should give us the same info as this type, but also mean that this logic doesn't need to be updated when we add new fields

Reply:
Honestly, the purposes were: _(a) remove noise_ -not sure if all fields make sense on the report-; and  _(b) beautify the output_ -for instance, by default the version is an array of two ints-, but yeah, sounds like `this logic doesn't need to be updated when we add new fields` is key here.

In addition, I guess part of the reason is because I'm not very familiar yet with different `kindsys` types, but also because I wasn't sure about specific details like `Meta.CurrentVersion` vs `thema.Lineage.Latest` 🙃

Anyway, I can definitely replace `kind` by `kindsys.SomeKindMeta`, yeah!

Reply:
Aaaaah, I forgot it! As part of "beautify the output", I also meant what's discussed here: https://github.com/grafana/grafana/pull/59742#discussion_r1039838709

Reply:
Yeah, i think not needing to update is probably the biggest deal - it's probably easier to ignore fields we don't care about in the dashboard than have to keep this up to date with the kindsys metas.